core/sql/bin/ex_ssmp_main.cpp

```diff
@@ -246,8 +247,12 @@ void runServer(Int32 argc, char **argv)
     }
   }
 */
-    // wait for system messages only until ssmp starts receiving msgs.
-    cc->wait(300);
+    // Wait for messages, but we need ssmp to wake up periodically to
+    // perform garbage collection.
+    short mask = XWAIT(LREQ | LDONE, ssmpGlobals->getStatsMergeTimeout());
+    if (mask & LREQ) {
+      cc->wait(0);
```

Comment:
This change looks good except that it is not encapsulated.  XWAIT, LREQ and LDONE concepts should have been part of the IPC infrastructure and not expected to be used in the callers of IPC infrastructure.  So, @narendragoyal and I tried the following  and it seems to achieve the same effect. 

Add the following 2 lines in lieu of this code
if (cc->getConnection() == NULL)
   cc->wait(ssmpGlobals->getStatsMergeTimeout());

And undo the change at ssmpGlobals::work().

Prior to Trafodion there used to at least one opener to ssmp always. Hence this issue was not observed earlier.  It is good catch

Reply:
Please consider modifying to encapsulate as suggested or some other way

Reply:
There is a subtle difference, the waitOnAll() function called in ssmpGlobals::work() does not wait on all connections simultaneously, it actually only waits on the first connection with an exponentially increased timeout (3 seconds maximum), and then cycle through all other connections without waiting. This can still lead to a unnecessary wait if there are data arrival on connections other than the first.

Reply:
WaitOnAll contains a set of connections : Control Connection + all other opener connections. Control connection can receive system messages like open, close etc, while other connection receive  messages sent by the openers. Though waitOnAll waits exponentially till the given timeout, as soon as any system message or messages from the opener is received it will break the  wait to respond to the message.

Reply:
It's OK to merge the PR as is now, we can do the encapsulation later, we may also need to do some refactoring and cleanup of the code, which may be risky and need more QA testing.